meos/src/general/temporal_modif.c

```diff
@@ -2110,15 +2110,20 @@ temporal_append_tsequence(Temporal *temp, const TSequence *seq, bool expand)
   {
     case TINSTANT:
     {
-      TSequence *seq1 = tinstant_to_tsequence((TInstant *) temp, interp2);
+      TSequence *seq1 = tinstant_to_tsequence((const TInstant *) temp, interp2);
       Temporal *result = (Temporal *) tsequence_append_tsequence(
-        (TSequence *) seq1, (TSequence *) seq, expand);
+        (const TSequence *) seq1, seq, expand);
       pfree(seq1);
+      pfree(temp);
       return result;
     }
     case TSEQUENCE:
-      return (Temporal *) tsequence_append_tsequence((TSequence *) temp, seq,
+    {
+      Temporal *result = tsequence_append_tsequence((const TSequence *) temp, seq,
         expand);
+      pfree(temp);
```

Comment:
The tests are now failing, I guess it's because of these added `free`s. Since these changes are not part of the objective of this PR, I'll remove them and create another PR if needed